### PR TITLE
ovl_Obj_Swprize OK

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c [hunk 1]
```diff
@@ -32,30 +35,113 @@ const ActorInit Obj_Swprize_InitVars = {
...
+    Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_RotateXS(this->actor.shape.rot.x, MTXMODE_APPLY);
+    Matrix_RotateZS(this->actor.shape.rot.z, MTXMODE_APPLY);
+    Matrix_MultVecY(1.0f, vec);
+    Matrix_Pop();
+}
+
+void func_80C253D0(ObjSwprize* this, PlayState* play) {
+    s32 i;
+    ObjSwprize* new_var;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c:53]: I think this is a thisx temp: https://decomp.me/scratch/IqyIx

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c [hunk 2]
```diff
@@ -32,30 +35,113 @@ const ActorInit Obj_Swprize_InitVars = {
...
+        new_var = this;
+        for (i = 0; i < 3; i++) {
+            collectible = Item_DropCollectible(play, &new_var->actor.world.pos, params);
+            if (collectible != NULL) {
+                if (sp78.y < 0.98f) {
+                    collectible->velocity.y = (sp78.y + 1.0f) * 4.0f;
+                    collectible->speedXZ = (2.0f * (1.0f - fabsf(sp78.y))) + 2.0f;
+                    collectible->world.rot.y = Math_FAtan2F(sp78.z, sp78.x) + D_80C257F8[i];
+                } else {
+                    collectible->world.rot.y = i * 0x5555;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c:70]:
```suggestion
                    collectible->world.rot.y = i * (0x10000 / 3);
```

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.h [hunk 3]
```diff
@@ -3,14 +3,17 @@
...
+
 struct ObjSwprize;
 
 typedef void (*ObjSwprizeActionFunc)(struct ObjSwprize*, PlayState*);
 
 typedef struct ObjSwprize {
     /* 0x000 */ Actor actor;
     /* 0x144 */ ObjSwprizeActionFunc actionFunc;
-    /* 0x148 */ char unk_148[0x4];
+    /* 0x148 */ s16 timer;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.h:16]:
```suggestion
    /* 0x148 */ s16 timer;
```

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c (hunk 1 above)
Comment (sonicdcer) [reply src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c:53]: nice catch, thank you!

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c [hunk 4]
```diff
@@ -32,30 +35,112 @@ const ActorInit Obj_Swprize_InitVars = {
...
+    Matrix_MultVecY(1.0f, vec);
+    Matrix_Pop();
+}
+
+void func_80C253D0(ObjSwprize* this, PlayState* play) {
+    Actor* thisx = &this->actor;
+    s32 i;
+    Actor* collectible;
+    Vec3f sp78;
+    s32 type = (thisx->params >> 8) & 3;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c:56]:
```suggestion
    s32 type = OBJ_SWPRIZE_GET_TYPE(thisx);
```

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c [hunk 5]
```diff
@@ -32,30 +35,112 @@ const ActorInit Obj_Swprize_InitVars = {
...
+    if (this->timer > 0) {
+        this->timer--;
+        if (this->timer == 0) {
+            ActorCutscene_Stop(this->actor.cutscene);
+            func_80C25780(this);
+        }
+    }
+}
+
+void func_80C25780(ObjSwprize* this) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c:135]:
```suggestion
void ObjSwprize_SetupDoNothing(ObjSwprize* this) {
```

File: src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c [hunk 6]
```diff
@@ -32,30 +35,113 @@ const ActorInit Obj_Swprize_InitVars = {
...
+    this->actionFunc = ObjSwprize_SetupDoNothing;
+}
+
+void ObjSwprize_SetupDoNothing(ObjSwprize* this, PlayState* play) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Swprize/z_obj_swprize.c:140]: Looks like you renamed the wrong function.

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Pull request merged**